packages/core/lib/command-utils.js

```diff
@@ -134,7 +139,11 @@ const prepareOptions = ({ command, inputStrings, options }) => {
   let invalidOptions = inputOptions.filter(opt => !validOptions.includes(opt));
 
   // TODO: Remove exception for 'truffle run' when plugin options support added.
-  if (invalidOptions.length > 0 && command.name !== "run") {
+  if (
+    invalidOptions.length > 0 &&
+    command.name !== "run" &&
+    command.name !== "version"
```

Comment:
Hmm, maybe it is better if we stick `--version` and `-v` in the allowed global options for the version command and make a comment about why it is there? I don't know, they are both hacks and I don't know which is better. Anyone else have opinions on this?

Reply:
Or you could maybe remove the `--version` or `-v` from the result that `extractFlags` returns in the case that we are running the version command. It's kind of ugly writing so much code for this exception but I'm not sure about what the alternative should be.

On the other hand you could just do the same exact thing we do with `--help` and `help` in `cli.js` and create a similar method to `displayGeneralHelp` that would allow us to avoid all this stuff.

Reply:
if we do a util function for displaying version info like `displayGeneralHelp` we would need to call this logic as well inside of the `main()` in `console-child.js`. This will require changes in two files. 

If we filter out `--v` out of the `inputOptions` (result of `extractFlags`), then only changes are on `command-util` but still in two functions `getCommand` and `prepareOptions`. 

Which one of these routes would be better?

Reply:
@lsqproduction, I think the better way might be to filter out the `--version` or `-v` from the `extractFlags` result as @eggplantzzz suggested.

Reply:
Yes, I also prefer this method. I will get this up.